CRM/Supportcase/Utils/Case.php

```diff
@@ -157,4 +157,79 @@ public static function findClientsIds($caseDataFromApi3) {
     return $clientIds;
   }
 
+  /**
+   * Gets case statuses with grouping "Closed"
+   *
+   * @return array
+   */
+  public static function getCaseClosedStatuses() {
+    try {
+      $caseStatuses = civicrm_api3('OptionValue', 'get', [
+        'sequential' => 1,
+        'option_group_id' => "case_status",
+        'grouping' => "Closed",
+        'is_active' => 1,
+        'options' => [
+          'limit' => 0,
+          'sort' => 'weight',
+        ],
+      ]);
+    } catch (CiviCRM_API3_Exception $e) {
+      return [];
+    }
+
+    return $caseStatuses['values'];
+  }
+
+  /**
+   * Gets values of case statuses with grouping "Closed"
+   *
+   * @return array
+   */
+  public static function getCaseClosedStatusesIds() {
+    $closedCaseStatuses = CRM_Supportcase_Utils_Case::getCaseClosedStatuses();
+    $caseClosedStatusesIds = [];
+
+    foreach ($closedCaseStatuses as $status) {
+      $caseClosedStatusesIds[] = $status['value'];
+    }
+
+    return $caseClosedStatusesIds;
+  }
+
+  /**
+   * Is case has a draft email?
+   *
+   * @return bool
+   */
+  public static function isCaseHasDraftEmails($caseId) {
+    try {
+      $result = civicrm_api3('SupportcaseDraftEmail', 'get', [
+        'case_id' => $caseId,
+      ]);
+    } catch (CiviCRM_API3_Exception $e) {
+      return false;
+    }
+
+    return !empty($result['values']);
+  }
+
+  /**
+   * Is case has a draft email?
+   *
+   * @return bool
+   */
+  public static function isCaseHasNotAnsveredEmail($caseId) {
+    try {
+      $result = civicrm_api3('Activity', 'get', [
```

Comment:
If I understand this correctly, this just looks for any activity of type "Inbound Email".
Instead, it should look for cases where there the most recent email activity is "Inbound Email" (= someone send us a message) and not "Email" (= we did not reply).